### DISCUSSION TITLE
Where is the database schema does the review (approve) flag/field live?

### DISCUSSION THREAD

**Discussion by AbhiOommen** [photoprism/photoprism#3497] [Q&A] [answered]
I have a ton of photos' (35,000+) to review and all of them are to be approved, using the GUI it will take a while to do this. 

I can write a quick UPDATE or INSERT statement to change the status of these pictures to approved, but after spending considerable amount of time I am unable to figure out which field in the database corresponds to that flag or whether there is a table for for approved photos to be added to, I did study the ERD on the site but couldn't figure it out, hence reaching out for help. 

Could someone please let me know where this field is located in the photoprism schema and what the relations are? Or if the approve process works by adding them to a different 'approved only' table when approved then let me know what table that is please?

Thanks.

**Comment by graciousgrey** [comment]
You can simply turn the review feature off in settings:

- https://docs.photoprism.app/user-guide/organize/review/
- https://docs.photoprism.app/user-guide/settings/library/#quality-filter

**Reply by AbhiOommen** [reply]
Yes, but then I lose the review functionality altogether. I have tried turning it off and back on, but then they all get back in review status when i turn it back on. Is there a way to prevent this? I added a bunch of old pictures to the library recently. I just want to be able to approve all of them for now and remove them from the review purgatory. I would like to retain the review functionality for the future.

**Reply by lastzero** [reply, marked as answer]
You could set the edited_at column to a (non NULL) timestamp and photo_quality to 3 for all quality values >= 0.

**Reply by AbhiOommen** [reply]
Thank You! the `photo_quality` filed name is what I was looking for.

**Comment by AbhiOommen** [comment]
> You could set the edit_at column to a (non NULL) timestamp and the quality to 3 for all quality values >= 0.

Don't you mean _for all quality values < 3_?

**Reply by lastzero** [reply]
-1 means the file is broken, this is different from having bad quality. Broken/hidden files show up in Library > Hidden.

**Reply by AbhiOommen** [reply]
Got it.  My main concern was mainly with the large number of items under _REVIEW_- .  So an update clause to force those quality scores from 0-2  up to 3 fixed that issue. no they are all reviewed.

As per the count I have about 82 photos with a -1 score which I will take a look at and see what is the issue.  Althoughtinterestingly nothing appears under Library > Hidden.

**Reply by lastzero** [reply]
Maybe deleted then?

**Reply by lastzero** [reply]
Make sure to set the edited_at timestamp so the photo_quality remains unchanged when you index the next time.

**Reply by AbhiOommen** [reply]
Yeah that makes sense then. Thank you once again. Very helpful.

**Reply by AbhiOommen** [reply]
> Make sure to set the edited_at timestamp so the photo_quality remains unchanged when you index the next time.

Yes.
```
UPDATE photos ph
SET ph.edited_at = NOW(), ph.photo_quality = 3
WHERE ph.photo_quality BETWEEN 0 AND 2;
```
